tests/test_configuration_auto.py

```diff
@@ -0,0 +1,44 @@
+# coding=utf-8
+# Copyright 2019-present, the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+
+from transformers.configuration_auto import AutoConfig
+from transformers.configuration_bert import BertConfig
+from transformers.configuration_roberta import RobertaConfig
+
+from .utils import DUMMY_UNKWOWN_IDENTIFIER
+
+
+SAMPLE_ROBERTA_CONFIG = os.path.join(os.path.dirname(os.path.abspath(__file__)), "fixtures/dummy-config.json")
+
+
+class AutoConfigTest(unittest.TestCase):
```

Comment:
I think we should add safety checks for the common errors we have had in `AutoXXX` classes here.

In particular we should check that the mapping dicts (`CONFIG_MAPPING` and the model/tokenizer mapping dicts) have:
- no key string that is included in a later key string (typical failure case), and
- no identical keys (that a silent failure in python) <= for this later case we should store the list used to create the `OrderedDict` in a separate module-level variable like `CONFIG_MAPPING_LIST` (before creation) of the `OrderedDict` and check that there are no duplicated keys.